users/twschum/twschum.c

```diff
@@ -0,0 +1,270 @@
+    case _Vim:
+        rgblight_sethsv_noeeprom(_Vim_HSV);
+        break;
+    case _Fn:
+        rgblight_sethsv_noeeprom(_Fn_HSV);
+        break;
+    case _Nav:
+        rgblight_sethsv_noeeprom(_Nav_HSV);
+        break;
+    case _Num:
+        rgblight_sethsv_noeeprom(_Num_HSV);
+        break;
+    case _Cfg:
+        rgblight_sethsv_noeeprom(_Cfg_HSV);
+        break;
+    case _None:
+        rgblight_sethsv_noeeprom(_None_HSV);
+        break;
+    }
+    return state;
+}
+#endif /* RGBLIGHT_ENABLE */
+
+/* process_record_vimlayer: handles the VIM_ keycodes from xtonhasvim's vim
+ * emulation layer
+ * add process_record_keymap to allow specific keymap to still add keys
+ * Makes the callstack look like:
+ * process_record_
+ *  _quantum
+ *    _kb
+ *      _user
+ *        _keymap
+ *        _vimlayer
+ */
+__attribute__ ((weak))
+bool process_record_keymap(uint16_t keycode, keyrecord_t *record) {
+  return true;
+}
+
+/* Return True to continue processing keycode, false to stop further processing
+ * process_record_keymap to be call by process_record_user in the vim addon */
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+
+  /* keymap gets first whack, then vimlayer */
+  if(!process_record_keymap(keycode, record)) return false;
+  if(!process_record_vimlayer(keycode, record)) return false;
+
+    switch (keycode) {
+        /* KC_MAKE is a keycode to be used with any keymap
+         * Outputs `make <keyboard>:<keymap>`
+         * Holding shift will add the appropriate flashing command (:dfu,
+         *   :teensy, :avrdude, :dfu-util) for a majority of keyboards.
+         * Holding control will add some commands that will speed up compiling
+         *   time by processing multiple files at once
+         * For the boards that lack a shift key, or that you want to always
+         *   attempt the flashing part, you can add FLASH_BOOTLOADER = yes to the
+         *   rules.mk of that keymap.
+         */
+        case KC_MAKE:  // Compiles the firmware, and adds the flash command based on keyboard bootloader
+            if (!record->event.pressed) {
+            uint8_t temp_mod = get_mods();
+            uint8_t temp_osm = get_oneshot_mods();
+            clear_mods(); clear_oneshot_mods();
+            SEND_STRING("make " QMK_KEYBOARD ":" QMK_KEYMAP);
+        #ifndef FLASH_BOOTLOADER
+            if ( (temp_mod | temp_osm) & MOD_MASK_SHIFT )
+        #endif
+            { //
+                #if defined(__arm__)  // only run for ARM boards
+                    SEND_STRING(":dfu-util");
+                #elif defined(BOOTLOADER_DFU) // only run for DFU boards
+                    SEND_STRING(":dfu");
+                #elif defined(BOOTLOADER_HALFKAY) // only run for teensy boards
+                    SEND_STRING(":teensy");
+                #elif defined(BOOTLOADER_CATERINA) // only run for Pro Micros
+                    SEND_STRING(":avrdude");
+                #endif // bootloader options
```

Comment:
All of this could be replaced with just: 
```
SEND_STRING(":flash");
```

Reply:
Oh awesome